src/api/spec/mixins/parse_package_diff_spec.rb

```diff
@@ -0,0 +1,156 @@
+require 'rails_helper'
+
+RSpec.describe ParsePackageDiff do
+  let(:instance_with_parse_package_diff_support) do
+    fake_instance = double("Fake Instance with ParsePackageDiff")
+    fake_instance.extend(ParsePackageDiff)
+    fake_instance
+  end
+
+  RSpec.shared_examples "package with a diff" do
+    let(:package_diff) do
+      "<sourcediff key='461472c75f0df9421a89f528417e72eb'>
+        <old project='home:Admin' package='test' rev='4' srcmd5='61c8de91f59df43c9ffd1fa9b4a3f055' />
+        <new project='home:Admin' package='test' rev='5' srcmd5='ca37dc90f6fd88f63db2ac9f1fc5c41c' />
+        <files>
+          <file state='changed'>
+            <old name='#{filename}' md5='f00a43bbe6d74b350577e5bce2ea5ff7' size='42' />
+            <new name='#{filename}' md5='3fd7513ed78f95e2be1bb211369bbea3' size='10' />
+            <diff lines='1'># the diff</diff>
+          </file>
+        </files>
+        <issues>
+        </issues>
+      </sourcediff>"
+    end
+
+    subject { instance_with_parse_package_diff_support.sorted_filenames_from_sourcediff(package_diff).first }
+
+    it { expect(subject['filenames']).to eq([filename]) }
+    it { expect(subject['old']).to eq({ "project" => "home:Admin", "package"=>"test", "rev"=>"4", "srcmd5"=>"61c8de91f59df43c9ffd1fa9b4a3f055" }) }
+    it { expect(subject['new']).to eq({ "project" => "home:Admin", "package"=>"test", "rev"=>"5", "srcmd5"=>"ca37dc90f6fd88f63db2ac9f1fc5c41c" }) }
+    it { expect(subject['files'][filename]['state']).to eq('changed') }
+    it { expect(subject['files'][filename]['old']).to eq({ "name" => filename, "md5" => "f00a43bbe6d74b350577e5bce2ea5ff7", "size" => "42" }) }
+    it { expect(subject['files'][filename]['new']).to eq({ "name" => filename, "md5" => "3fd7513ed78f95e2be1bb211369bbea3", "size" => "10" }) }
+    it { expect(subject['files'][filename]['diff']).to eq({ "lines" => "1", "_content" => "# the diff" }) }
+  end
+
+  describe '#sorted_filenames_from_sourcediff' do
+    context 'a other file' do
+      let(:filename) { 'Dockerfile' }
+      it_behaves_like  "package with a diff"
+    end
+
+    context 'a file with a /' do
+      let(:filename) { '/foobar' }
+      it_behaves_like  "package with a diff"
+    end
+
+    context 'a spec file' do
+      let(:filename) { 'apache2.spec' }
+      it_behaves_like  "package with a diff"
+    end
+
+    context 'a dif file' do
+      let(:filename) { 'apache2.dif' }
+      it_behaves_like  "package with a diff"
+    end
+
+    context 'a diff file' do
+      let(:filename) { 'apache2.diff' }
+      it_behaves_like  "package with a diff"
+    end
+
+    context 'a patch file' do
+      let(:filename) { 'apache2.patch' }
+      it_behaves_like  "package with a diff"
+    end
```

Comment:
Does it actually make sense to test the various file suffixes, if this just get's copied over to the xml in the test?

Reply:
Dropped as it is already covered in the "more than one file" case. The important part to test is that the filenames are ordered...